File: Practice(Beginner)/Python3/Forgotten_Language.py
# Source: https://www.codechef.com/problems/FRGTNLNG

t = int(input())
for test in range(t):
    n, k = map(int, input().split())
    lang = list(map(str,input().split()))
    s = set()
    for i in range(k):
        l, *temp = map(str,input().split())
        s.update(set(temp))
    s = set(s)
    for j in lang:
        if (j in s):
            print("YES", end=" ")
        else:
            print("NO", end=" ")
    print()

File: Practice(Beginner)/C++/Forgotten Language.cpp
# Source: https://www.codechef.com/problems/FRGTNLNG

#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, n, k, l;
    string temp;
    cin >> t;
    while (t--) {
        cin >> n >> k;
        vector<string> v;
        for (int i = 0; i < n; ++i) {
            cin >> temp;
            v.push_back(temp);
        }
        set<string> s;
        while (k--){
            cin >> l;
            for (int i = 0; i < l; ++i) {
                cin >> temp;
                s.insert(temp);
            }
        }
        for (int i = 0; i < n; ++i) {
            auto it = s.find(v[i]);
            if(it != s.end()){
                cout << "YES ";
            }else{
                cout << "NO ";
            }
        }
        cout << endl;
    }
    return 0;
}